desertbot/modules/commands/OutOfContext.py

```diff
@@ -0,0 +1,244 @@
+            return IRCResponse(ResponseType.Say, "That message is not in the log.", message.replyTo)
+        if len(matches) > 1:
+            return IRCResponse(ResponseType.Say,
+                               f"Unable to remove quote, {len(matches)} matches found.",
+                               message.replyTo)
+        return self._removeQuote(message.replyTo, matches[0])
+
+    def _removebyid(self, message: IRCMessage):
+        """removebyid <quoteid> - remove the quote with the specified ID from the OutOfContext log."""
+        if len(message.parameterList) < 2:
+            return IRCResponse(ResponseType.Say, "Remove what?", message.replyTo)
+        if not string.isNumber(message.parameterList[1]):
+            return IRCResponse(ResponseType.Say, "You didn't specify a valid ID.", message.replyTo)
+        if len(self.storage) == 0 or message.replyTo not in self.storage:
+            return IRCResponse(ResponseType.Say, "There are no quotes in the log.", message.replyTo)
+        index = int(message.parameterList[1]) - 1
+        quotes = self.storage[message.replyTo]
+        if index < len(quotes):
+            return self._removeQuote(message.replyTo, quotes[index])
+        return IRCResponse(ResponseType.Say, "That message is not in the log.", message.replyTo)
+
+    def _list(self, message: IRCMessage):
+        """list (<search/searchnick <regex>>) - post the OutOfContext log. A search regex can be provided to filter\
+         the list."""
+        params = [x for x in message.parameterList[1:]]
+        if len(params) > 0:
+            subsubcommand = params.pop(0).lower()
+            if subsubcommand == "searchnick":
+                return self._postList(message.replyTo, " ".join(params), True)
+            elif subsubcommand == "search":
+                return self._postList(message.replyTo, " ".join(params), False)
+            else:
+                return self._postList(message.replyTo, "", False)
+        else:
+            return self._postList(message.replyTo, "", False)
+
+    def _search(self, message: IRCMessage):
+        """search <regex> - look up quotes in the OutOfContext log. This search operation will look at \
+        the content of the quotes."""
+        if len(message.parameterList) < 2:
+            return IRCResponse(ResponseType.Say, "Search what?", message.replyTo)
+        return self._getQuote(message.replyTo, " ".join(message.parameterList[1:]), False, -1)
+
+    def _searchnick(self, message: IRCMessage):
+        """searchnick <regex> - look up quotes in the OutOfContext log. This search operation will \
+        look at the nick that said the quoted line."""
+        nick = message.user.nick
+        if len(message.parameterList) > 1:
+            nick = message.parameterList[1]
+        return self._getQuote(message.replyTo, nick, True, -1)
+
+    def _id(self, message: IRCMessage):
+        """oocid <quoteid> - look up the quote that has the given ID."""
```

Comment:
Should this helptext not just be id and not oocid? The subcommand is ".ooc id" no?